### PR TITLE
Add preparation of images as part of RC preparation process

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
I enabled preparing RC images as part of the release process. I think it might be useful for people to run some tests for RCs more easily - especially if they use docker-compose or helm chart.

**Comment by potiuk** [comment]
@jhtimmins - you will have to ask (via INFRA Jira ticket) so that you are added to "airflow admins" group for in DockerHub - to be able to push images there.

**Comment by potiuk** [comment]
Looks like all good. Would be handy for the final release of 2.1.1 :). The nice thing is that we can use master scripts + Docker to build the image from the released version.

**Comment by ashb** [comment]
Can he ask or would we (PMC) have to ask on his behalf?

File: scripts/ci/libraries/_build_images.sh [hunk 1]
```diff
@@ -743,8 +743,8 @@ function build_images::prepare_prod_build() {
         build_images::add_build_args_for_remote_install
     elif [[ -n "${INSTALL_AIRFLOW_VERSION=}" ]]; then
         # When --install-airflow-version is used then the image is build using released PIP package
-        # For PROD image only numeric versions are allowed
-        if [[ ! ${INSTALL_AIRFLOW_VERSION} =~ ^[0-9\.]*$ ]]; then
+        # For PROD image only numeric versions are allowed and RC candidates
+        if [[ ! ${INSTALL_AIRFLOW_VERSION} =~ ^[0-9\.]*(rc[0-9]*)?$ ]]; then
```
Comment (ashb) [inline scripts/ci/libraries/_build_images.sh]:
```suggestion
        if [[ ! ${INSTALL_AIRFLOW_VERSION} =~ ^[0-9\.]+(rc[0-9]+)?$ ]]; then
```

Comment (ashb) [reply scripts/ci/libraries/_build_images.sh]: (Assuming bash regex work this way)

File: tests/jobs/test_scheduler_job.py [hunk 2]
```diff
@@ -3650,7 +3650,7 @@ def test_task_with_upstream_skip_process_task_instances():
     assert dr is not None
 
     with create_session() as session:
-        tis = {ti.task_id: ti for ti in dr.get_task_instances(session=session)}
+        tis = {ti.task_id: ti for ti in dr.get_task_instances(session=session)}  # pylint: disable=no-member
```
Comment (ashb) [inline tests/jobs/test_scheduler_job.py]: Not needed now.

**Comment by potiuk** [comment]
> Can he ask or would we (PMC) have to ask on his behalf?

I asked as a committer and they gave me access, so I guess should be possible :). But I am happy to create a ticket there if I know the username on DockerHub.

File: scripts/ci/libraries/_build_images.sh (hunk 1 above)
Comment (potiuk) [reply scripts/ci/libraries/_build_images.sh]: Ah indeed.

File: tests/jobs/test_scheduler_job.py (hunk 2 above)
Comment (potiuk) [reply tests/jobs/test_scheduler_job.py]:
```suggestion
        tis = {ti.task_id: ti for ti in dr.get_task_instances(session=session)}
```

Comment (potiuk) [reply tests/jobs/test_scheduler_job.py]: True :)

File: tests/jobs/test_scheduler_job.py [hunk 3]
```diff
@@ -3660,7 +3660,7 @@ def test_task_with_upstream_skip_process_task_instances():
     # dag_file_processor._process_task_instances(dag, dag_runs=dag_runs)
 
     with create_session() as session:
-        tis = {ti.task_id: ti for ti in dr.get_task_instances(session=session)}
+        tis = {ti.task_id: ti for ti in dr.get_task_instances(session=session)}  # pylint: disable=no-member
```
Comment (ashb) [inline tests/jobs/test_scheduler_job.py]:
```suggestion
        tis = {ti.task_id: ti for ti in dr.get_task_instances(session=session)}
```

**Review by ashb — APPROVED** [review]

**Pull request merged**